File: ModeratorServer/src/main/java/dk/groupfive/ModeratorServer/model/objects/ReviewFull.java
package dk.groupfive.ModeratorServer.model.objects;

import java.util.ArrayList;
import java.util.List;

public class ReviewFull implements IReview {
    private ArrayList<ReviewItem> reviews;

    public ReviewFull() {
        reviews = new ArrayList<>();
    }

    @Override
    public double getRating() {
        long score = 0;
        for (ReviewItem item : reviews)
        {
            score += item.getRating();
        }
        return (double) score / reviews.size();
    }

    @Override
    public List<ReviewItem> getReviews() {
        return reviews;
    }
}

File: DataServer/DataServer/Model/ReviewFull.cs
using DataServer.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace DataServer.Models
{
	class ReviewFull : IReview
	{
		public List<ReviewItem> reviews { get; set; }

		public ReviewFull()
		{
			reviews = new List<ReviewItem>();
		}

		public double GetRating()
		{
			long score = 0;
			foreach (ReviewItem item in reviews)
			{
				score += item.rating;
			}
			return (double)score / reviews.Count;
		}

		public List<ReviewItem> GetReviews()
		{
			return reviews;
		}

		public void AddReview(ReviewItem review)
		{
			reviews.Add(review);
		}
	}
}
